### PR TITLE
Beforeinput prevent default on mismatching targetRanges

### LINKED ISSUE #3953: Bug: On Mac, period is append to token nodes with auto punctuation feature enabled
On Mac, we noticed that “.” is appended incorrectly to token nodes after typing double-space (with the Add period with double-space feature enabled). Is this a known issue? I can repro on both Lexical playground and the WP comment composer. It recovers itself when the editor state is updated thought. The workaround seems to be removing the character in updateDOM, but we are wondering if there is a better solution

Lexical version: I can repro on the Lexical playground

## Steps To Reproduce

1. With auto punctuation enabled on Mac (https://user-images.githubusercontent.com/118264/220735887-41877324-a70c-4cfe-ab4e-84355d4c9a67.png)

2. Follow the steps shown in the video below
https://user-images.githubusercontent.com/118264/220735615-38fc0a3e-ba2a-4872-a3a7-37a3bf25bea6.mp4

Link to code example:

### PR DESCRIPTION
**Pull request by zurfyx**
The "Add period with double-space after emoji" Mac feature misbehaves when triggered immediately after a node on Chrome. My hunch is that this is a Chrome-specific bug, as Safari does it correcty.

However, what brought me to this fix is the fact that this can lead to a more generic issue. We allow anyting to happen on the DOM on beforeinput. In this case, Mac is saying insert `. ` which covers the immediately previous 2 position but the targetRange could be completely arbitrary, overriding some of our other DOM elements.

Given that this is rare, it's worth switching to controlled mode when this happens and it will have no impact whatsoever on the performance.

Alternatively, we could handle it in the MutationObserver, trying to intercept what has been modified and what-not but given that MutationObserver triggers after beforeinput (when we handle the input) this overcomplicates the solution for an edge case for a neglible performance gain.

Fixes #3953

### CODE REVIEW

File: packages/lexical-playground/__tests__/e2e/Events.spec.mjs [hunk 1]
```diff
@@ -99,4 +98,80 @@ test.describe('Events', () => {
       `,
     );
   });
+
+  test('Add period with double-space after emoji (MacOS specific) #3953', async ({
```
[the remark this answers is not in the document]
Comment (zurfyx) [reply packages/lexical-playground/__tests__/e2e/Events.spec.mjs:102]: Think it's fine to run on everything. I actually have strong opinions against the regression tests folder. It's much more visual to have everything sorted by topic than some random tests in the regressions folder. I would much rather have these regressions annotations with the task name on the test itself. (let's discuss this in one of our syncs maybe)
